src/editor/Duplicate.js

```diff
@@ -2,58 +2,168 @@
+
+	useEffect(() => {
+		setError( null );
+		if ( modalVisible && ! duplicateTitle ) {
+			getDefaultDupicateTitle();
+		}
+		if ( ! modalVisible ) {
+			setDuplicated( null );
+			setDuplicateTitle( null );
+		}
+	}, [ modalVisible ]);
+
+	const getDefaultDupicateTitle = async () => {
+		const promptToDuplicate = parseInt( duplicateOf || postId );
+		try {
+			const defaultTitle = await apiFetch( {
+				path: `/newspack-popups/v1/${ promptToDuplicate }/duplicate`,
+			} );
+
+			setDuplicateTitle( defaultTitle );
+		} catch ( e ) {
+			setDuplicateTitle( title + __( ' copy', 'newspack-popups' ) );
+		}
+	};
+
+	const duplicatePrompt = async ( popupId, titleForDuplicate ) => {
+		setError( null );
 		try {
 			const newId = await apiFetch( {
-				path: `/newspack-popups/v1/${ postId }/duplicate`,
+				path: addQueryArgs( `/newspack-popups/v1/${ popupId }/duplicate`, {
+					title: titleForDuplicate,
+				} ),
 				method: 'POST',
 			} );
 
 			if ( isNaN( newId ) ) {
 				throw new Error( __( 'Error duplicating prompt.', 'newspack-popups' ) );
 			}
 
-			createNotice( 'success', __( 'Prompt duplicated! Redirecting...', 'newspack-popups' ), {
-				id: 'newspack-popups__duplicate-prompt-success',
-				isDismissible: false,
-				type: 'default',
-			} );
-
 			// Redirect to edit page for the copy.
-			window.location = `/wp-admin/post.php?post=${ newId }&action=edit`;
+			setDuplicated( newId );
 		} catch ( e ) {
-			createNotice( 'error', e?.message || __( 'Error duplicating prompt.', 'newspack-popups' ), {
-				id: 'newspack-popups__duplicate-prompt-error',
-				isDismissible: true,
-				type: 'default',
-			} );
+			setError( e?.message || __( 'Error duplicating prompt.', 'newspack-popups' ) );
 		}
 	};
 
 	return (
-		<Button
-			isSecondary
-			isBusy={ isSavingPost }
-			disabled={ isSavingPost }
-			onClick={ () => autosave().then( () => duplicatePrompt() ) }
-		>
-			{ __( 'Duplicate', 'newspack-popups' ) }
-		</Button>
+		<>
+			<Button
+				isSecondary
+				isBusy={ isSavingPost }
+				disabled={ isSavingPost }
+				onClick={ () => setModalVisible( true ) }
+			>
+				{ __( 'Duplicate', 'newspack-popups' ) }
+			</Button>
+			{ modalVisible && (
+				<Modal
+					className="newspack-popups__duplicate-modal"
+					title={ sprintf( __( 'Duplicate “%s”', 'newspack-popups' ), title ) }
+					onRequestClose={ () => setModalVisible( false ) }
+				>
+					{ error && (
+						<Notice isDismissible={ false } status="error">
+							{ error }
+						</Notice>
+					) }
+					{ duplicated ? (
+						<>
+							<Notice status="success" isDismissible={ false }>
+								{ sprintf(
+									__( 'Duplicate of “%s” created as a draft.', 'newspack-popups' ),
+									title
+								) }
+							</Notice>
+							{ ( ! campaignGroups || 0 === campaignGroups.length ) && (
+								<Notice status="warning" isDismissible={ false }>
+									{ __( 'The prompt is currently unassigned.', 'newspack-popups' ) }
```

Comment:
WDYT think about demystifying "unassigned" a bit:

```suggestion
									{ __( 'This prompt is currently not assigned to any campaign.', 'newspack-popups' ) }
```